packages/quantic/force-app/main/default/lwc/quanticCaseClassification/e2e/quanticCaseClassification.e2e.ts

```diff
@@ -0,0 +1,269 @@
+        caseAssist.waitForCaseClassificationsResponse();
+
+      await caseClassification.clickShowSelectInputButton();
+      await caseClassification.clickallOptionsSelectInput();
+      await caseClassification.clickSelectInputOptionAtIndex(3);
+
+      await caseClassificationsResponsePromise;
+    });
+  });
+
+  test.describe('when the property fetchDocumentSuggestionOnChange is set to true', () => {
+    test.use({
+      options: {fetchDocumentSuggestionOnChange: true},
+    });
+
+    test('should trigger a new document suggestions request when a case classification suggestion is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const documentSuggestionResponsePromise =
+        caseAssist.waitForDocumentSuggestionResponse();
+
+      await caseClassification.clickCaseClassificationSuggestionAtIndex(1);
+
+      await documentSuggestionResponsePromise;
+    });
+
+    test('should trigger a new document suggestions request when a case classification option is clicked', async ({
+      caseClassification,
+      caseAssist,
+    }) => {
+      const documentSuggestionResponsePromise =
+        caseAssist.waitForDocumentSuggestionResponse();
+
+      await caseClassification.clickShowSelectInputButton();
+      await caseClassification.clickallOptionsSelectInput();
+      await caseClassification.clickSelectInputOptionAtIndex(3);
+
+      await documentSuggestionResponsePromise;
+    });
+  });
+
+  test.describe('the behaviour of the case classifications', () => {
+    test.describe('when selecting a suggestion and then receiving new suggestions that include the selected suggestion', () => {
+      test('should keep the suggestion by the user selected and display it as an inline option', async ({
+        caseClassification,
+        caseAssist,
+      }) => {
+        const selectedSuggestionIndex = 1;
+        await caseClassification.clickCaseClassificationSuggestionAtIndex(
+          selectedSuggestionIndex
+        );
+        await expect(caseClassification.selectedSuggestion).toHaveValue(
+          caseClassificationSuggestions[selectedSuggestionIndex].value
+        );
+
+        caseClassification.mockCaseClassification(
+          coveoDefaultField,
+          caseClassificationSuggestions
+        );
+        const caseClassificationsResponsePromise =
+          caseAssist.waitForCaseClassificationsResponse();
+        await caseAssist.fetchClassifications();
+        await caseClassificationsResponsePromise;
+        await expect(caseClassification.selectedSuggestion).toHaveValue(
+          caseClassificationSuggestions[selectedSuggestionIndex].value
+        );
+      });
+    });
+
+    test.describe('when selecting a specific suggestion and then receiving new suggestions that does not contain the previously selected option', () => {
+      test('should keep the suggestion selected by the user and display it in the select input', async ({
```

Comment:
This also could be a unit test I think?

Reply:
this is not a unit tested because it would require mocking the subscribe mechanism of the headless engine which risk the unit test file complicated.

additionally, this for me fall in the interactions that the user is doing with the component, since the user here is selecting a suggestion then triggering a new case classification request (in real life this would happen when the user re-updates his Case subject for example in his case assist flow).

Reply:
Yea okay, I was more thinking that it might be easier to do if we simply would be mocking the state and then triggering an event :) all good to keep it then.

It's true it was a user interaction that is the "trigger" but I saw this as a specific situation of the state, more than a specific user interaction that would've triggered this test.

Reply:
The thing that additionally makes this a bit difficult to test in unit test is that this behaviour is controlled solely by the state of the Quantic component and not by the Headless controller, so we can't just set the Headless state to whatever we want and be able to test this situation.
internal logic of this component is what leads you to this situation I am testing here.